service/src/main/java/bio/terra/workspace/service/workspace/flight/CreateWorkspaceStep.java

```diff
@@ -53,7 +54,16 @@ public StepResult doStep(FlightContext flightContext)
             .description(description)
             .build();
 
-    workspaceDao.createWorkspace(workspaceToCreate);
+    try {
+      workspaceDao.createWorkspace(workspaceToCreate);
+    } catch (DuplicateWorkspaceException ex) {
+      // This might be the result of a step re-running, or it might be an ID conflict. We can ignore
+      // this if the existing workspace matches the one we were about to create, otherwise rethrow.
+      Workspace existingWorkspace = workspaceDao.getWorkspace(workspaceId);
```

Comment:
Since we generate the workspace ID as a random UUID, can we ever really get an ID conflict that is not the step re-running?

Not that you should remove the check; defense in depth is good. But maybe amend the comment?